client/pages/projects/main-section.js

```diff
@@ -1,10 +1,29 @@
+import ProjectComponent from "/components/project-component.js";
+
 const MainSection = {
-  data: () => ({ message: "Projects" }),
+  components: {
+    ProjectComponent,
+  },
   template: `
-    <div class="main">
-      <h1>{{message}}</h1>
-    </div>
-    `,
+  <div class="project__banner--container">
+    <h2 class="project__banner--title">Discover all the <span style="color:white">projects</span> that have been built by HYF students</h2>
+    <img src="../../images/projects/project-page-banner.png" alt="project image" class="project__banner--image">
+  </div>
+    <template v-for="project in projects">
+      <project-component :project="project"/>
+    </template>
+  </div>
+  `,
+  data() {
+    return {
+      projects: [
```

Comment:
in my members page for the members array which here you are using projects array in the data function
I declared an empty array then after calling the users data from the backend (fetchUsers) I used the fetched array data to push into the members array 
for example: 
```
data(){
return {
members:[],
};
},
methods: {
async onload(){
try {
const result await fetchUsers();
for (let i=0; i< result.length; i++){
this.members.push({
username: result[i].FirstName + " " + result[i].LastName,
avatar: result[i].ProfilePicture,
title: result[i].JobTitle,
ClassID: result[i].ClassID,
})
}
}
}}
} catch (error) {
 console.log("error from members", error);
}
```
if you want you can take this part and change the namings to have the data you need to present into the frontend from the backend
I hope my point is clear to you

Reply:
Thanks @Moamin-AbuEwaida I have the idea.

Reply:
I used some faked values to show the results. I shouldn't have used them

Reply:
I wanted to show the frontend results to be visible in the server first because I know it will take me some time to finish the backend part, at least that's what I think